service/src/test/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceTest.java

```diff
@@ -322,57 +334,142 @@ private BaseResource getResource() {
     @DataProvider(name = "filterPolicySetsInvalidRequestDataProvider")
     private Object[][] filterPolicySetsInvalidRequestDataProvider()
             throws JsonParseException, JsonMappingException, IOException {
-        return new Object[][] {
-                { createOnePolicySet(), Arrays.asList(new String[] { "Invalid policy set id" }) },
-                { createTwoPolicySets(), Arrays.asList(new String[] {}) },
-                { createTwoPolicySets(), Arrays
-                        .asList(new String[] { "policy-set-with-one-policy-one-condition", "noexistent-policy-set" }) } };
+        List<PolicySet> onePolicySet = createOnePolicySet();
+        List<PolicySet> twoPolicySets = createTwoPolicySets();
+        return new Object[][] { filterOnePolicySetByNonexistentPolicySet(onePolicySet),
+                filterTwoPolicySetsByNull(twoPolicySets), filterTwoPolisySetsByEmptyList(twoPolicySets),
+                filterTwoPolicySetsByByNonexistentPolicySet(twoPolicySets) };
+    }
+
+    private Object[] filterOnePolicySetByNonexistentPolicySet(List<PolicySet> onePolicySet) {
+        return new Object[] { onePolicySet, Arrays.asList(new String[] { "nonexistent-policy-set" }) };
+    }
+
+    private Object[] filterTwoPolicySetsByNull(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, null };
+    }
+
+    private Object[] filterTwoPolisySetsByEmptyList(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, Collections.emptyList() };
+    }
+
+    private Object[] filterTwoPolicySetsByByNonexistentPolicySet(List<PolicySet> twoPolicySets) {
+        return new Object[] { twoPolicySets, Arrays
+                .asList(new String[] { twoPolicySets.get(0).getName(), "noexistent-policy-set" }) };
     }
 
     @DataProvider(name = "filterPolicySetsDataProvider")
-    private Object[][] filterPolicySetsDataProvider() throws JsonParseException, JsonMappingException, IOException {
+    private Object[][] filterPolicySetsDataProvider() {
         List<PolicySet> onePolicySet = createOnePolicySet();
         List<PolicySet> twoPolicySets = createTwoPolicySets();
-        return new Object[][] {
-            { onePolicySet, null, onePolicySet },
-            { onePolicySet, Collections.emptyList(), onePolicySet },
-            { onePolicySet, Arrays.asList(onePolicySet.get(0).getName()), onePolicySet },
-            { twoPolicySets, Arrays.asList(twoPolicySets.get(0).getName()), twoPolicySets.subList(0, 1) },
-            { twoPolicySets, Arrays.asList(twoPolicySets.get(1).getName()), twoPolicySets.subList(1, 2) },
-            { twoPolicySets,
-                Arrays.asList(twoPolicySets.get(0).getName(), twoPolicySets.get(1).getName()), twoPolicySets }
+        return new Object[][] { filterNullPolicySetsListByNull(),
```

Comment:
General comment on this class and null handling:  (null checks are an [anti-pattern](http://stackoverflow.com/questions/271526/avoiding-null-statements).  Use spring's Assert instead of java assert keyword though.  )

- PolicyEvalService should assert, very early, that PolicySets and PolicySetsPriority are non-null.  (PolicyService and Controller must ensure this)
- This will remove null checks from the service and null v/s empty tests in unit tests